Src/HALALMock/Services/SharedMemory/SharedMemory.cpp

```diff
@@ -1,9 +1,60 @@
 
 #include "HALALMock/Services/SharedMemory/SharedMemory.hpp"
+//includes to create the shared Memory in Posix
+#include <fcntl.h>
+#include <sys/shm.h>
+#include <sys/stat.h>
+#include <sys/mman.h>
+#include <unistd.h>
 
 void SharedMemory::start() {
     // TODO: create / open shared memory
 }
+
+std::string SharedMemory::generate_shared_memory_name(){
+	return "SharedMemory_emulated_state_machine_"+std::to_string(getpid());
+}
+
+void SharedMemory::start_emulated_state_machine(){
+	// shared memory file descriptor
+	int shm_state_machine_fd;
+	emulated_state_machine_name=generate_shared_memory_name();
+
+	// create the shared memory object
+	shm_state_machine_fd=shm_open(emulated_state_machine_name.c_str(),O_CREAT | O_RDWR, 0666);
+	if(shm_state_machine_fd==-1){
+		std::cout<<"Error creating the shared memory object\n";
+		std::terminate();
+	}
+	// configure the size of the shared memory object
+	if(ftruncate(shm_state_machine_fd,emulated_state_machine_size)==-1){
+		std::cout<<"Error configuring the size of the shared memory object\n";
+		close(shm_state_machine_fd);
+		std::terminate();
+	}
+	// memory map the shared memory object
+	emulated_state_machine=mmap(NULL,emulated_state_machine_size,PROT_WRITE | PROT_READ,MAP_SHARED,shm_state_machine_fd,0);
+	if(emulated_state_machine==MAP_FAILED){
+		std::cout<<"Error mapping the shared memory object\n";
+		close(shm_state_machine_fd);
+		std::terminate();
+	}
+
+	// cast of the void pointer that the shared memory object returns to a uint8_t pointer
+	state_machine_sm=static_cast<uint8_t*>(emulated_state_machine);
+	// initialize the shared memory counter by setting it to 0, when a new state machine is created it will increment
+	state_machine_sm[0]=0;
```

Comment:
oh, this will crash your program, because when it gets initialized it gets the value `nullptr`, which you cannot derefence, if you want to keep this logic just make a call to `malloc()` specifying some buffer so that you are sure it is your memory, ideally use unique_ptr so you free the memory whenever the program ends.

Reply:
(for @g0nz4I0) You are mixing things, this pointer comes from the mmap call, it has just been casted to `uint8_t*`, so it will be valid, no need to call malloc because all that memory allocation has already been performed by shm_open, ftruncate and mmap

Reply:
xd, I dont know what  I read, honestly I though the pointer wasn't even initialized, my bad.